yt/data_objects/tests/test_profiles.py

```diff
@@ -285,8 +273,16 @@ def DM_in_cell_mass(field, data):
 
     add_particle_filter("DM", function=DMparticles,
                         filtered_type='io', requires=["particle_type"])
-
-    ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030")
+    _fields = ("particle_position_x", "particle_position_y",
+               "particle_position_z", "particle_mass", "particle_velocity_x",
+               "particle_velocity_y", "particle_velocity_z", "particle_type",
+               "density", "temperature")
+    _units = ('cm', 'cm', 'cm', 'g', 'cm/s', 'cm/s', 'cm/s', 'dimensionless',
+              'g/cm**3','K')
+    _negative = (False, False, False, False, True, True, True, False, False,
+                 False)
+    ds = fake_particle_ds(fields=_fields, units=_units, negative=_negative,
+                          npart=16 ** 2, length_unit=1.0)
```

Comment:
I think you need to use `fake_random_ds` here, but tell it to include particle fields as well. By using a purely particle dataset here you're changing what this test is actually doing under the hood. It needs to be run using a dataset with both mesh and particle fields because the profile that gets created below gets binned using two mesh fields `('gas', 'temperature')` and `('gas', 'density')`, but that actual field that's being profiled is a deposited particle field. So you need both a mesh to deposit onto and particle fields to deposit onto the mesh.

Let me know if you run into trouble getting `fake_random_ds` working.

Reply:
I need help in implementing this test case with `fake_random_ds`.

Reply:
Take a look at this example:

```
In [3]: ds = fake_random_ds(32, particle_fields=("particle_position_x", "particle_position_y", "particle_position_z", "particle_mass", "particle_velocity_x", "particle_velocity_y", "particle_velocity_z"), particle_field_units=('cm', 'cm', 'cm', 'g', 'cm/s', 'cm/s', 'cm/s'), particles=16)

In [4]: ds.particle_type_counts
Out[4]: {'io': 16}

In [5]: ad = ds.all_data()

In [6]: ad['gas', 'density']
Out[6]:
YTArray([0.76017901, 0.96855994, 0.49205428, ..., 0.78097504, 0.61756868,
         0.27913556]) g/cm**3

In [7]: ad['gas', 'density'].shape
Out[7]: (32768,)

In [8]: ad['deposit', 'io_cic']
Out[8]: YTArray([0., 0., 0., ..., 0., 0., 0.]) g/cm**3

In [9]: ad['deposit', 'io_cic'].shape
Out[9]: (32768,)

In [11]: ad['deposit', 'io_mass'].sum()
Out[11]: 8.105905659452372 g

In [12]: ad['io', 'particle_mass'].sum()
Out[12]: 8.105905659452372 g

Reply:
thanks, pushed the changes.